src/plugins/workspace/public/components/workspace_form/workspace_form.tsx

```diff
@@ -0,0 +1,222 @@
+          })}
+          isInvalid={!!formErrors.name}
+          error={formErrors.name}
+        >
+          <EuiFieldText
+            value={formData.name}
+            onChange={handleNameInputChange}
+            readOnly={!!defaultValues?.reserved}
+            data-test-subj="workspaceForm-workspaceDetails-nameInputText"
+          />
+        </EuiFormRow>
+        <EuiFormRow
+          label={
+            <>
+              Description - <i>optional</i>
+            </>
+          }
+          helpText={i18n.translate('workspace.form.workspaceDetails.description.helpText', {
+            defaultMessage:
+              'Valid characters are a-z, A-Z, 0-9, (), [], _ (underscore), - (hyphen) and (space).',
+          })}
+          isInvalid={!!formErrors.description}
+          error={formErrors.description}
+        >
+          <EuiFieldText
+            value={formData.description}
+            onChange={handleDescriptionInputChange}
+            data-test-subj="workspaceForm-workspaceDetails-descriptionInputText"
+          />
+        </EuiFormRow>
+        <EuiFormRow
+          label={i18n.translate('workspace.form.workspaceDetails.color.label', {
+            defaultMessage: 'Color',
+          })}
+          isInvalid={!!formErrors.color}
+          error={formErrors.color}
+        >
+          <div>
+            <EuiText size="xs" color="subdued">
+              {i18n.translate('workspace.form.workspaceDetails.color.helpText', {
+                defaultMessage: 'Accent color for your workspace',
+              })}
+            </EuiText>
+            <EuiSpacer size={'s'} />
+            <EuiColorPicker
+              color={formData.color}
+              onChange={handleColorChange}
+              data-test-subj="workspaceForm-workspaceDetails-colorPicker"
+            />
+          </div>
+        </EuiFormRow>
+        <EuiFormRow
+          label={i18n.translate('workspace.form.workspaceDetails.icon.label', {
+            defaultMessage: 'Icon',
+          })}
+          isInvalid={!!formErrors.icon}
+          error={formErrors.icon}
+        >
+          <WorkspaceIconSelector
+            value={formData.icon}
+            onChange={handleIconChange}
+            color={formData.color}
+          />
+        </EuiFormRow>
+        <EuiFormRow
+          label={i18n.translate('workspace.form.workspaceDetails.defaultVisualizationTheme.label', {
+            defaultMessage: 'Default visualization theme',
+          })}
+          isInvalid={!!formErrors.defaultVISTheme}
+          error={formErrors.defaultVISTheme}
+        >
+          <EuiSelect
+            hasNoInitialSelection
+            value={formData.defaultVISTheme}
+            options={defaultVISThemeOptions}
+            onChange={handleDefaultVISThemeChange}
+            data-test-subj="workspaceForm-workspaceDetails-defaultVISThemeSelector"
+          />
+        </EuiFormRow>
```

Comment:
I may suggest to remove vis theme, color and icon when it's unclear how these will be used.